templates/content/node--full.html.twig

```diff
@@ -90,10 +89,16 @@
 <article{{ attributes.addClass(classes) }}>
 
   {{ title_prefix }}
-  {% if label and not page %}
-    <h2{{ title_attributes }}>
-      <a href="{{ url }}" rel="bookmark">{{ label }}</a>
-    </h2>
+  {% if label %}
+    {% if not page %}
+      <h2{{ title_attributes }}>
+        <a href="{{ url }}" rel="bookmark">{{ label }}</a>
+      </h2>
+    {% else %}
+      <h1{{ title_attributes.addClass('page-title') }}>
+        {{ label }}
+      </h1>
+    {% endif %}
   {% endif %}
```

Comment:
This is an unlikely condition since it's the `node--full.html.twig` template but I wanted to have the logic represented in code somewhere, in case the template is copied for other node view mode templates.
I also didn't want to have a `node.html.twig` template because it seems too generic and thus unuseful, but I can introduce one if the conditional title heading logic is better suited there for clarity.